templates/hooks/certificate/sdk_create_post_request.go.tpl

```diff
@@ -0,0 +1,17 @@
+    if err != nil {
+		input := &svcsdk.DescribeCertificateAuthorityInput{}
+		input.CertificateAuthorityArn = desired.ko.Spec.CertificateAuthorityARN
+
+		var describeResp *svcsdk.DescribeCertificateAuthorityOutput
+		describeResp, describeErr := rm.sdkapi.DescribeCertificateAuthorityWithContext(ctx, input)
+		rm.metrics.RecordAPICall("READ_ONE", "DescribeCertificateAuthority", err)
+		if describeErr != nil {
+			return desired, ackrequeue.NeededAfter(describeErr, ackrequeue.DefaultRequeueAfterDuration)
+		}
+
+		if *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusFailed && *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDeleted && *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDisabled {
+			return desired, ackrequeue.NeededAfter(err, ackrequeue.DefaultRequeueAfterDuration)
+		}
+
+		return nil, err
+	}
```

Comment:
Why do we need to requeue if a CA is in FAILED state? will it eventually self recover?

Reply:
We want the Certificate and CertificateAuthorityActivation resources to requeue while the corresponding CA resource is not in a FAILED/DISABLED/DELETED state. This way a CA hierarchy can be created by applying one file and each resource can be synced, after waiting for the CA resource to be created.